eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -673,7 +673,8 @@ public function moveSubtree(APILocation $location, APILocation $newParentLocatio
                 'new parent Location is in a subtree of the given $location'
             );
         }
-        if (!$newParentLocation->getContent()->getContentType()->isContainer) {
+        $contentTypeId = $newParentLocation->contentInfo->contentTypeId;
+        if (!$this->repository->getContentTypeService()->loadContentType($contentTypeId)->isContainer) {
```

Comment:
We should not use `$this->repository->getContentTypeService` here but inject service direct or SPI to avoid circular references

Reply:
@mikadamczyk resolved in: https://github.com/ezsystems/ezpublish-kernel/pull/3074/commits/f797a7c0bd1521bac00fd3e26bff902bc3567f40